lib/user.rb

```diff
@@ -0,0 +1,41 @@
+require 'pg'
+require 'bcrypt'
+
+class User
+
+  attr_reader :id, :name, :email, :username
+
+  def initialize(id:, name:, email:, username:)
+    @id = id
+    @name = name
+    @email = email
+    @username = username
+  end
+
+  def self.create(name:, email:, username:, password:)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+    # encrypt the plantext password
+    encrypted_password = BCrypt::Password.create(password)
+
+    # insert the encrypted password into the database, instead of the plaintext one
+    result = connection.exec("INSERT INTO users (name, email, username, password) VALUES('#{name}', '#{email}', '#{username}', '#{encrypted_password}') RETURNING id, name, email, username;")
+    User.new(id: result[0]['id'], name: result[0]['name'], email: result[0]['email'], username: result[0]['username'])
+  end
+
+  def self.find(id)
+    return nil unless id
```

Comment:
Not sure that this line is doing anything.  there's always going to be an id - if there isn't then the method will throw an argument error.